qt/python/mantidqt/mantidqt/widgets/codeeditor/multifileinterpreter.py

```diff
@@ -403,3 +447,32 @@ def zoom_out_all_tabs(self):
     def on_completion_change(self):
         for idx in range(self.editor_count):
             self.editor_at(idx).setCompletion(self.completion_enabled)
+
+    def focus_in(self, file):
+        self.check_file_for_external_modifications(file)
+
+    def check_file_for_external_modifications(self, file):
+        if file not in self.files_changed_unhandled:
+            return
+
+        self.files_changed_unhandled.remove(file)
+        if osp.isfile(file):
+            reload_button = QMessageBox.question(
+                self,
+                "",
+                f"The current file ({file}) has been modified by an external source. \n\nWould you like to reload the file?",
+                buttons=(QMessageBox.Yes | QMessageBox.No),
+                defaultButton=QMessageBox.Yes,
+            )
+            if reload_button == QMessageBox.Yes:
+                self.reload_file(file)
+            else:
+                self.current_editor().mark_as_modified()
+        else:
+            QMessageBox.warning(
+                self,
+                "",
+                f"The current file ({file}) doesn't exist anymore, which is due to an external source deleting or renaming the \
+                    file. The contents of this tab will remain unchanged, but are unsaved.",
+            )
```

Comment:
Could we add another sentence which explains that saving will bring back / recreate the file